logstash-core/lib/logstash/api/commands/stats.rb

```diff
@@ -5,17 +5,32 @@ module LogStash
   module Api
     module Commands
       class Stats < Commands::Base
-
         def jvm
-          {:threads => service.get_shallow(:jvm, :threads)}
+          {
```

Comment:
This was refactored to use `extract_metrics` to make the mapping of metrics to field names more explicit. This way we can add metrics internally to a namespace without necessarily exposing them to the API.